techniques/system/common/1.0/promises.st

```diff
@@ -190,7 +190,7 @@ bundle agent check_disable_agent
 
       # Only define this class when we're ready to die - this is a special class name in "abortclasses"
       "should_not_continue"
-        expression => "should_disable_agent",
+        expression => "should_disable_agent.!ignore_disable_agent",
```

Comment:
Well, "rudder agent inventory" already uses the force_inventory class. I think that the name is explicit enough that it should always run an inventory. I see no plausible case where you'd run "cf-agent -D force_inventory" and expect to be informed inventories are disabled.